apps/arena/lib/arena/game_updater.ex

```diff
@@ -911,6 +937,49 @@ defmodule Arena.GameUpdater do
     end
   end
 
+  defp handle_pools(%{players: players} = game_state, game_config) do
+    Enum.reduce(players, game_state, fn {_player_id, player}, game_state ->
+      Enum.reduce(game_state.pools, game_state, fn {pool_id, pool}, acc ->
```

Comment:
I think we can avoid this second reduce by just taking the pools we collided with (player.collides_with) and then adding those effects and removing the others.

Reply:
The thing is that we need to know certainly which effects from the player we should remove, so we need to know if a player collides with a pool or not